### ISSUE TITLE
hangs when linking Template Haskell 8.0/8.2

### ISSUE THREAD

**Issue by lspitzner** [ghcjs/ghcjs#668]
E.g. when building aeson-1.2.4.0 [edit: same for aeson versions 1.2.0.0, 1.3.0.0, 1.4.0.0] with both ghcjs-8.0 and 8.2, the last lines of output are:

~~~~
[18 of 24] Compiling Data.Aeson.Types.ToJSON ( Data/Aeson/Types/ToJSON.hs, dist/
  dist-sandbox-94af1e4/build/Data/Aeson/Types/ToJSON.js_o )                                                                             
Linking Template Haskell (Data.Aeson.Compat,Data.Aeson.Encoding,Data.Aeson.Encod
  ing.Builder,Data.Aeson.Encoding.Internal,Data.Aeson.Internal,Data.Aeson.Intern
  al.Functions,Data.Aeson.Internal.Time,Data.Aeson.Parser,Data.Aeson.Parser.Inte
  rnal,Data.Aeson.Parser.Time,Data.Aeson.Parser.Unescape,Data.Aeson.Parser.Unesc
  apePure,Data.Aeson.Types.FromJSON,Data.Aeson.Types.Generic,Data.Aeson.Types.In
  ternal,Data.Attoparsec.Time,Data.Attoparsec.Time.Internal,ThRunner1)     
~~~~

at which point there is a `node thrunner.js` process running, but it seems to hang (<0.01 cpu usage, constant memory <200MB). I have killed the process after ~30 minutes of it doing nothing.

~~~~
> ghcjs --version
The Glorious Glasgow Haskell Compilation System for JavaScript, version 8.2.0.1 (GHC 8.2.2)
> node --version
v10.3.0
~~~~

I don't think this is #449, because the reports there imply a lot of cpu/ram usage.

~~I have not tested downgrading nodejs as it seems kind of annoying to do without downgrading other system stuff as well. If this is indeed nodejs-dependent, I'd be thankful for an upper bound or any other hint regarding supported nodejs versions.~~

Oh, and in case this has a different cause, my system is a mostly up-to-date archlinux.

**Comment by colbyn** [comment]
I have encountered the exact same issue; while building a project for an interview 😭...

**Comment by lspitzner** [comment]
One more data-point: node v7.10.1 works. This makes it very likely that some change in node causes this.

**Comment by luite** [comment]
I can reproduce this with the latest version. node v8.9.4 was ok, v10.13.0 hangs (linux x64)

**Comment by luite** [comment]
should be fixed by dc190b1bb2453cfa484124e9f335ee3cad1492f7

**Comment by luite** [comment]
(but this deprecation warning shouldn't have interfered with the template Haskell session. using a different pipe for the TH communication should be done later)
